airbyte-integrations/connectors/source-bing-ads/source_bing_ads/spec.json

```diff
@@ -69,6 +69,64 @@
         "minimum": 0,
         "maximum": 90,
         "order": 6
+      },
+      "custom_reports": {
+        "title": "Custom Reports",
+        "description": "You can add your Custom Bing Ads report by creating one.",
+        "order": 7,
+        "type": "array",
+        "items": {
+          "title": "Custom Report Config",
+          "type": "object",
+          "properties": {
+            "name": {
+              "title": "Report Name",
+              "description": "The name of the custom report, this name would be used as stream name",
+              "type": "string",
+              "examples": [
+                "Account Performance",
+                "AdDynamicTextPerformanceReport",
+                "custom report"
+              ]
+            },
+            "reporting_object": {
+              "title": "Reporting Data Object",
+              "description": "The name of the the object derives from the ReportRequest object. You can find it in Bing Ads Api docs - Reporting API - Reporting Data Objects.",
+              "type": "string",
+              "examples": [
+                "AccountPerformanceReport",
+                "AdDynamicTextPerformanceReport"
+              ]
+            },
+            "report_columns": {
+              "title": "Columns",
+              "description": "A list of available report object columns. You can find it in description of reporting object that you want to add to custom report.",
```

Comment:
Please provide the root link where the columns can be seen and taken, so the customer can easily follow the link and check them out.

Reply:
all reporting data objects have different columns, so the link relates to data object that was specified by user

Reply:
Yes, unfortunately, this is the limitation for the dependent dropdowns currently, it's fine to leave as is, but we need to provide the link that could describe the columns available for each data object and aggregation type. It's better to mention this info in the public docs for the user.